src/foraging_gui/GenerateMetadata.py

```diff
@@ -0,0 +1,758 @@
+            'camera_list':['SideCameraLeft','SideCameraRight','BottomCamera','BodyCamera'], # camera names in the settings_box.csv
+            'camera_name_mapper':{
+                'SideCameraLeft': "Face side left",
+                'SideCameraRight': "Face side right",
+                'BottomCamera': "Bottom",
+                'BodyCamera': "Body"
+            }, # camera names in the settings_box.csv and the corresponding names in the rig metadata
+        }
+
+    def _get_box_type(self):
+        '''
+        To judge the box type (ephys or behavior) based on the rig_id. This should be improved in the future.
+        '''
+
+        if 'rig_id' not in self.Obj['meta_data_dialog']['rig_metadata']:
+            self.box_type = 'Unknown'
+            return
+        
+        if 'EPHYS' in self.Obj['meta_data_dialog']['rig_metadata']['rig_id']:
+            self.box_type = 'Ephys'
+        else:
+            self.box_type = 'Behavior'
+    
+    def _session_description(self):
+        '''
+        Generate the session description to the MetadataFolder
+        '''
+        if self.Obj['meta_data_dialog']['rig_metadata']=={}:
+            return
+        self._get_session_time()
+        if self.session_start_time == '' or self.session_end_time == '':
+            return
+        
+        self.orcid = BaseName(name="Open Researcher and Contributor ID", abbreviation="ORCID")
+        self._get_modality()
+        self._get_investigators()
+        self._get_funding_source()
+        self._get_platform()
+
+        description= RawDataDescription(
+            data_level=DataLevel.RAW,
+            funding_source=self.funding_source,
+            investigators=self.investigators,
+            modality=self.modality,
+            project_name=self.Obj['meta_data_dialog']['session_metadata']['ProjectName'],
+            data_summary=self.Obj['meta_data_dialog']['session_metadata']['DataSummary'],
+            institution=Organization.AIND,
+            creation_time=self.session_start_time,
+            platform= self.platform,
+            subject_id=self.Obj['ID'],
+        )
+        description.write_standard_file(output_directory=self.Obj['MetadataFolder'])
+
+    def _get_funding_source(self):
+        '''
+        Get the funding source
+        '''
+        self.funding_source=[Funding(
+            funder=getattr(Organization,self.Obj['meta_data_dialog']['session_metadata']['FundingSource']),
+            grant_number=self.Obj['meta_data_dialog']['session_metadata']['GrantNumber'],
+            fundee=self.Obj['meta_data_dialog']['session_metadata']['Fundee'],
+        )]
+                
+    def _get_platform(self):
+        '''
+        Get the platform name. This should be improved in the future.
+        '''
+        if self.box_type == 'Ephys':
+            self.platform = Platform.ECEPHYS
```

Comment:
I think we made the decision that the platform is always "behavior" regardless of whether there is ephys data. (its the way the folders are named)

Reply:
I am not sure. It makes more sense to me to differentiate between EPHYS and BEHAVIOR platforms. We have different camera setups and have more devices used in the ephys rig.

Reply:
I'm saying this decision has already been made in favor of `behavior`. I understand the arguments in favor of the other decision

Reply:
Anna is still using the EPHYS platform in the 323_EPHYS1. Who made this decision?

Reply:
Anna is running behavior?

Reply:
Updated here: https://github.com/AllenNeuralDynamics/dynamic-foraging-task/pull/299

Reply:
Made issue: https://github.com/AllenNeuralDynamics/dynamic-foraging-task/issues/469